From cdbff1bcdc28631983ae6384ed9897927e03d4b6 Mon Sep 17 00:00:00 2001
From: Mohamed Ashiq <mliyazud@redhat.com>
Date: Thu, 5 Nov 2015 20:22:02 +0530
Subject: [PATCH 145/162] afr:vol heal info fails when transport.socket.bind-address is set in glusterd

Problem: If glusterd's bind address is specified using transport.socket.bind- address
option to a specific IP, the heal command 'gluster volume heal <VOLNAME> info' fails
with error '<vol_name>: Not able to fetch volfile from glusterd.Volume heal failed.'
This patch uses Unix domain socket transport type of libglfapi to get the volfile.

>Change-Id: Ic4239b339f0246d77f1622c07d5f4708129d30c7
>BUG: 1277997
>Signed-off-by: Mohamed Ashiq <mliyazud@redhat.com>
>Signed-off-by: Humble Devassy Chirammal <hchiramm@redhat.com>
>Reviewed-on: http://review.gluster.org/12508
>Tested-by: Gluster Build System <jenkins@build.gluster.com>
>Reviewed-by: Ravishankar N <ravishankar@redhat.com>
>Tested-by: NetBSD Build System <jenkins@build.gluster.org>
>Reviewed-by: Niels de Vos <ndevos@redhat.com>
>Reviewed-by: Anuradha Talur <atalur@redhat.com>
>Reviewed-by: Pranith Kumar Karampuri <pkarampu@redhat.com>

Change-Id: Ifda39831c0fc0d55174c61eea563a406992198eb
BUG: 1257343
Signed-off-by: Mohamed Ashiq Liyazudeen <mliyazud@redhat.com>
Signed-off-by: Humble Devassy Chirammal <hchiramm@redhat.com>
Reviewed-on: https://code.engineering.redhat.com/gerrit/62580
Reviewed-by: Ravishankar Narayanankutty <ravishankar@redhat.com>
Tested-by: Ravishankar Narayanankutty <ravishankar@redhat.com>
---
 heal/src/glfs-heal.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/heal/src/glfs-heal.c b/heal/src/glfs-heal.c
index 207e472..2f120c9 100644
--- a/heal/src/glfs-heal.c
+++ b/heal/src/glfs-heal.c
@@ -19,6 +19,7 @@
 #include "syncop-utils.h"
 #include <string.h>
 #include <time.h>
+#include "glusterfs.h"
 
 #define DEFAULT_HEAL_LOG_FILE_DIRECTORY DATADIR "/log/glusterfs"
 #define USAGE_STR "Usage: %s <VOLNAME> [bigger-file <FILE> | "\
@@ -756,7 +757,7 @@ main (int argc, char **argv)
                 goto out;
         }
 
-        ret = glfs_set_volfile_server (fs, "tcp", "localhost", 24007);
+        ret = glfs_set_volfile_server (fs, "unix", DEFAULT_GLUSTERD_SOCKFILE, 0);
         if (ret) {
                 printf("Setting the volfile server failed, %s\n", strerror (errno));
                 goto out;
-- 
1.7.1

